pkg/api/v1/atlasdatabaseuser_types.go

```diff
@@ -140,6 +143,15 @@ type AtlasDatabaseUserList struct {
 	Items           []AtlasDatabaseUser `json:"items"`
 }
 
+// ReconciliableRequests implements ReconciliableList for Database Users
+func (list *AtlasDatabaseUserList) ReconciliableRequests() []reconcile.Request {
```

Comment:
I suggest to leave this in the indexer function. This function is part of `pkg/api` which is official API surface. We should keep it clean, just having structs and metadata for kubebuilder.

Reply:
We would have a problem with circular dependency, wouldn't we? Also, this is likely to be used by the indexes mapper function which usually lives in the controller.
Last, I'm wondering if a sugar abstraction to avoid iterating over list items is worth it.

Reply:
I need to think how to solve this. I understand that it should not be here. My goal was to avoid to have to write this logic for every new CR that would use the local creds.

Reply:
I am using a generic function instead. It is implemented for each supporting CRD on the indexer package, so not exposed any more.